### ISSUE TITLE
Support other operating systems

### ISSUE THREAD

**Issue by turicas** [srstevenson/xdg#3]
It would be awesome if this library supports other OSes, so we can use only one library to deal with user paths, no matter the OS the user is running.

I think 3 main OSes should be supported:

- GNU and other free/libre Unices (already working since they use XDG)
- Mac OS X (it's UNIX-like but does not follow XDG specs)
- Windows

Some links to help the discussion:

- [XDG support on neovim](https://github.com/neovim/neovim/issues/78)
- [XDG-equivalents on Mac OS X](https://stackoverflow.com/a/5084892/1299446)

**Comment by srstevenson** [comment]
`xdg` should already support those three operating systems; are there any particular issues you encountered @turicas? `xdg` follows the specification strictly, which doesn't for example specify different behaviour on Windows or macOS (such as using `%APPDATA%` and`~/Library` respectively); thus far I've considered these other conventions out of scope.

**Comment by mihaipopescu** [comment]
Yes, it would be ideal to have cross platform support just because I think it's the general expected behavior for a python package.

Otherwise, I'd have to resort to doing the cross platform support in my package.

As for my particular issue is that I'm calling asking for 
`xdg.XDG_CACHE_HOME` and I get `$HOME\.cache` which means that the home directory is not expanded to `%USERPROFILE%` as expected.

**Comment by mihaipopescu** [comment]
We could probably just use https://docs.python.org/2.7/library/os.path.html#os.path.expanduser, seems globally available.

**Comment by mentalisttraceur** [comment]
Opinion/thought:

So... an important point here is that strictly following the XDG specification on Windows or MacOS would create a Wrong app. (In the same sense of "wrong" as NOT following the XDG specification is "wrong" on Linux/BSD systems where users expect the XDG layout.)

For example, if I'm writing a GUI for end users who aren't used to Linux/UNIX conventions (most users), I would be unpleasantly/frustratingly surprising many of my users - they might expect their app stuff to be saved in AppData or otherwise out-of-view - instead they get some random folders like .config, .local, and so on junking up their user profile root - and of course on Windows dotfiles aren't hidden (whereas folders like AppData are).

None of this is to say that you "have to" support it in this package, of course. It's your package and if you want it narrowly scoped to just-XDG rather than XDG-or-equivalent-native-convention then that's fine.

But if I was writing an app or library that needed any of these directories, I wouldn't use a library that was only-XDG unless I knew my code was only going to be used on Linux/UNIX (and given how *thin* this library is, if I was releasing a more general XDG-or-equivalent library to the public, I would hesitate to include this as a dependency, because I know library consumers often prefer libraries which don't include a dependency for too little, I wouldn't be saving much effort/code, and the indirection wouldn't clarify/simplify/deduplicate my code enough to pay for itself).

So... just something to consider.

**Comment by mentalisttraceur** [comment]
I guess on MacOS things are a little less black-and-white - there's some argument that CLI tools on MacOS should just follow XDG specs.

But speaking the native language

**Comment by mentalisttraceur** [comment]
I guess on MacOS things are a little less black-and-white - there's some argument that CLI tools on MacOS should just follow XDG specs.

But speaking the "native language" of a system is often helpful in permeating, not-immediately-obvious ways, or even non-deterministically becomes relevant depending on user settings or after the system gets a new feature.

For example, I just found out that on MacOS, `~/Library/Caches` is automatically excluded from the "Time Machine" (native system-wide backup and version control) feature. That's something I just wouldn't've known if I hadn't stumbled upon it in the StackOverflow link @turicas posted, but that's something that I think a user would reasonably expect by default. Maybe XDG eventually wins and MacOS starts excluding `~/.cache` from backups too, but it's an example of how systems usually have various features and assumptions and third-party software relying on the native conventions being followed.

(Which is more reason to think that writers of libraries/GUIs would/should be more interested in something that's XDG-or-native-convention than just-XDG.)

Again, it's your library so stick with just-XDG if that's what feels right/happy for you - I'm just thinking out-loud more than anything else.

**Comment by mentalisttraceur** [comment]
Ah, nvm. I've remembered/learned that [`platformdirs`](https://pypi.org/project/platformdirs/) is very popular (carrying on where `appdirs` left off) and doing a fine job of filling the XDG-or-equivalent-directories niche.

In light of that, I think it's actually *good* that this package exists  in parallel and fills the just-XDG niche.
